lib/sequent/rake/tasks.rb

```diff
@@ -0,0 +1,104 @@
+require 'active_record'
+require 'rake'
+require 'rake/tasklib'
+
+require 'sequent/support'
+
+module Sequent
+  module Rake
+    class Tasks < ::Rake::TaskLib
+      include ::Rake::DSL
+
+      DEFAULT_OPTIONS = {
+        migrations_path: 'db/migrate',
+        event_store_schema: 'public'
+      }
+
+      attr_reader :options
+
+      def initialize(options)
+        @options = DEFAULT_OPTIONS.merge(options)
+      end
+
+      def register!
+        register_db_tasks!
+        register_view_schema_tasks!
+      end
+
+      def register_db_tasks!
+        namespace :db do
+          desc 'Create the database'
+          task :create do
+            current_environments.each do |env|
+              env_db = db_config(env)
+              puts "Create database #{env_db['database']}"
+              Sequent::Support::Database.create!(env_db)
+            end
+          end
+
+          desc 'Drop the database'
+          task :drop do
+            current_environments.each do |env|
+              env_db = db_config(env)
+              puts "Drop database #{env_db['database']}"
+              Sequent::Support::Database.drop!(env_db)
+            end
+          end
+
+          task :establish_connection do
+            env_db = db_config(options.fetch(:environment))
+            Sequent::Support::Database.establish_connections(env_db, view_projection.schema_name)
+          end
+
+          desc 'Migrate the database'
+          task migrate: :establish_connection do
+            database.create_schema_if_not_exists!(options.fetch(:event_store_schema))
+            database.migrate(options.fetch(:migrations_path))
+          end
+        end
+      end
+
+      def register_view_schema_tasks!
+        namespace :view_schema do
+          desc 'Build the view schema'
+          task build: :'db:establish_connection' do
+            if database.schema_exists?(view_projection.schema_name)
+              puts "View version #{view_projection.version} already exists; no need to build it"
+            else
+              database.create_schema!(view_projection.schema_name)
+              view_projection.build!
+            end
+          end
+
+          desc 'Drop the view schema'
+          task drop: :'db:establish_connection' do
+            database.drop_schema!(view_projection.schema_name)
+          end
+        end
+      end
+
+      private
+
+      def current_environments
+        envs = [options.fetch(:environment)]
+        envs << 'test' if @env == 'development'
```

Comment:
I wondered about this too for a while. It's what Rails does. It is helpful to maintain the test database without having to set an environment (so that after `db:create` and `db:migrate` ActiveRecord can be used in a spec helper to maintain the test schema).

If it's only confusing, we should remove it :)